index.js

```diff
@@ -373,11 +367,19 @@ class WriteStreamSync extends WriteStream {
   }
 }
 
-const writev = (fd, iovec, pos, cb) => {
-  const done = (er, bw) => cb(er, bw, iovec)
-  const req = new FSReqWrap()
-  req.oncomplete = done
-  binding.writeBuffers(fd, iovec, pos, req)
+let writev = fs.writev
+/* istanbul ignore next */
+if (!writev) {
+  const binding = process.binding('fs')
+  const writeBuffers = binding.writeBuffers
+  const FSReqWrap = binding.FSReqWrap || binding.FSReqCallback
+
+  writev = (fd, iovec, pos, cb) => {
+    const done = (er, bw) => cb(er, bw, iovec)
+    const req = new FSReqWrap()
+    req.oncomplete = done
+    binding.writeBuffers(fd, iovec, pos, req)
+  }
```

Comment:
nit: I realise by the time this all executes `let writev` will be populated and the class which is defined above will be able to use this file scoped variable, but _stylistically_ I think this block should be defined at the top under `const fs = require('fs')` and before `const _autoClose = Symbol('_autoClose')`

Reply:
OK, sure, moved!